### PR TITLE
implement originating origin identity

### PR DESCRIPTION
**Pull request by shawn-hurley**
* added a interface to the context
* Exposing the concrete classes for casting if wanted

### CODE REVIEW

**Comment by shawn-hurley** [comment]
I updated based on the comments and I also took a stab at adding a test case for making sure the Originating Origin Identity is being set.

File: pkg/rest/apisurface.go [hunk 1]
```diff
@@ -374,5 +398,33 @@ func unpackUpdateRequest(r *http.Request) (*osb.UpdateInstanceRequest, error) {
...
+	identity, err := retrieveOriginatingIdentity(r)
+	if err != nil {
+		return nil, err
+	}
+	osbRequest.OriginatingIdentity = identity
+
 	return osbRequest, nil
 }
+
+func retrieveOriginatingIdentity(r *http.Request) (*osb.OriginatingIdentity, error) {
```
Comment (pmorie) [inline pkg/rest/apisurface.go]: Will you add a godoc comment to this method?

File: pkg/server/deprovision_test.go [hunk 2]
```diff
@@ -74,6 +75,20 @@ func TestDeprovision(t *testing.T) {
 				Async: true,
 			},
 		},
+		{
+			name: "deprovision passes originating origin idenity is passed",
```
Comment (pmorie) [inline pkg/server/deprovision_test.go]: name sounds a little off

**Comment by pmorie** [comment]
LGTM, thanks @shawn-hurley

**Pull request merged**